### ISSUE TITLE
Empty schemas allow any options

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#4789]
Originally discovered by @gyandeeps in [#4786 (comment)](https://github.com/eslint/eslint/pull/4786#issuecomment-166641992).

```js
module.exports.schema = [];
```

Right now, the above schema will allow any number and kind of options after the severity, when it should only allow a severity.

Working on this.
